### PR TITLE
#205 Fix : handles ConnectionError

### PR DESCRIPTION
**Pull request by aalbatross**
Handled ConnectionError.
ran the local tests without new FAILURES.

### CODE REVIEW

File: bigchaindb/commands/bigchaindb.py [hunk 1]
```diff
@@ -153,7 +156,6 @@ def run_init(args):
         print('The database already exists.', file=sys.stderr)
         print('If you wish to re-initialize it, first drop it.', file=sys.stderr)
 
-
```
Comment (vrde) [inline bigchaindb/commands/bigchaindb.py]: Those `\n\n` are there because we follow [PEP8](https://www.python.org/dev/peps/pep-0008/#blank-lines).

**Comment by ttmc** [comment]
Hi @aalbatross 

Before we can merge this pull request, we need you or your organization to agree to one of our contributor agreements. One of the big concerns for people using and developing open source software is that someone who contributed to the code might claim the code infringes on their copyright or patent. To guard against this, we ask all our contributors to sign a Contributor License Agreement. This gives us the right to use the code contributed and any patents the contribution relies on. It also gives us and our users comfort that they won't be sued for using open source software. We know it's a hassle, but it makes the project more reliable in the long run. Thank you for your understanding and your contribution!

If you are contributing on behalf of yourself (and not on behalf of your employer or another organization you are part of) then you should:

1. Go to: https://www.bigchaindb.com/cla/
2. Read the Individual Contributor Agreement
3. Fill in the form "For Individuals"
4. Check the box to agree
5. Click the SEND button

If you're contributing as an employee, and/or you want all employees of your employing organization to be covered by our contributor agreement, then someone in your organization with the authority to enter agreements on behalf of all employees must do the following:

1. Go to: https://www.bigchaindb.com/cla/
2. Read the Entity Contributor Agreement
3. Fill in the form "For Organizations”
4. Check the box to agree
5. Click the SEND button

We will email you (or your employer) with further instructions.

**Comment by aalbatross** [comment]
@ttmc i have accepted CLA. @vrde will update the PR with test.

**Comment by ttmc** [comment]
@aalbatross Please copy the block of text that I emailed you to the comments in this pull request.

**Comment by aalbatross** [comment]
BEGIN BLOCK

This is to confirm that I agreed to and accepted the BigchainDB Individual Contributor Agreement at https://www.bigchaindb.com/cla/ and to represent and warrant that I have authority to do so.

9gga4Agz9jJT91bjR58Abi6lTbQIHtvmAqsvDb6yJ33RWbo4fi2u7zL3TdVJwFH

END BLOCK

**Comment by ttmc** [comment]
I approved because the CLA step is complete.

**Comment by vrde** [comment]
Hi @aalbatross, any updates on this PR? Thanks!

**Comment by aalbatross** [comment]
Hi vrde,

Working on writing testcase. Will update it asap.

Thanks
Ravi


On 02-Aug-2017 6:51 PM, "vrde" <notifications@github.com> wrote:

Hi @aalbatross <https://github.com/aalbatross>, any updates on this PR?
Thanks!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/bigchaindb/bigchaindb/pull/1694#issuecomment-319669989>,
or mute the thread

**Comment by aalbatross** [comment]
@vrde 
I added the testcase, but the test didnot PASS because Failed: DID NOT RAISE <class 'bigchaindb.backend.exceptions.ConnectionError'>

Since test runs with rethinkdb server started. How can i stop it while running the test. or can i mock it ? If yes how ?

**Comment by vrde** [comment]
> Since test runs with rethinkdb server started. How can i stop it while running the test. or can i mock it ? If yes how ?

Yes you can mock it. I guess something like [this](https://github.com/bigchaindb/bigchaindb/blob/99b477776608dbd30282b27753d6ec057d7752f0/tests/backend/mongodb/test_changefeed.py#L140) should work.

Moreover, there are other places where a connection can fail, like `run_drop`.

**Comment by vrde** [comment]
Since there is no further development, I'll close this PR.

**Pull request closed without merging**